processor/routingprocessor/routing.go

```diff
@@ -0,0 +1,182 @@
+	if len(oCfg.Table) == 0 {
+		return nil, fmt.Errorf("invalid routing table: %w", errNoTableItems)
+	}
+
+	// we also need a "FromAttribute" value
+	if len(oCfg.FromAttribute) == 0 {
+		return nil, fmt.Errorf("invalid attribute to read the route's value from: %w", errNoMissingFromAttribute)
+	}
+
+	return &processorImp{
+		logger:         logger,
+		config:         *oCfg,
+		traceExporters: make(map[string][]component.TraceExporter),
+	}, nil
+}
+
+func (e *processorImp) Start(_ context.Context, host component.Host) error {
+	// first, let's build a map of exporter names with the exporter instances
+	source := host.GetExporters()
+	availableExporters := map[string]component.TraceExporter{}
+	for k, exp := range source[configmodels.TracesDataType] {
+		traceExp, ok := exp.(component.TraceExporter)
+		if !ok {
+			return fmt.Errorf("the exporter %q isn't a trace exporter", k.Name())
+		}
+		availableExporters[k.Name()] = traceExp
+	}
+
+	// default exporters
+	if err := e.registerExportersForDefaultRoute(availableExporters, e.config.DefaultExporters); err != nil {
+		return err
+	}
+
+	// exporters for each defined value
+	for _, item := range e.config.Table {
+		if err := e.registerExportersForRoute(item.Value, availableExporters, item.Exporters); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (e *processorImp) registerExportersForDefaultRoute(available map[string]component.TraceExporter, requested []string) error {
+	for _, exp := range requested {
+		v, ok := available[exp]
+		if !ok {
+			return fmt.Errorf("error registering default exporter %q: %w", exp, errExporterNotFound)
+		}
+		e.defaultTraceExporters = append(e.defaultTraceExporters, v)
+	}
+
+	return nil
+}
+
+func (e *processorImp) registerExportersForRoute(route string, available map[string]component.TraceExporter, requested []string) error {
+	for _, exp := range requested {
+		v, ok := available[exp]
+		if !ok {
+			return fmt.Errorf("error registering route %q for exporter %q: %w", route, exp, errExporterNotFound)
+		}
+		e.traceExporters[route] = append(e.traceExporters[route], v)
+	}
+
+	return nil
+}
+
+func (e *processorImp) Shutdown(context.Context) error {
+	return nil
+}
+
+func (e *processorImp) ConsumeTraces(ctx context.Context, td pdata.Traces) error {
+	value := e.extractValueFromContext(ctx)
+	if len(value) == 0 {
+		// the attribute's value hasn't been found, send data to the default exporter
+		return e.pushDataToExporters(ctx, td, e.defaultTraceExporters)
+	}
+
+	if _, ok := e.traceExporters[value]; !ok {
+		// the value has been found, but there are no exporters for the value
+		return e.pushDataToExporters(ctx, td, e.defaultTraceExporters)
+	}
+
+	// found the appropriate router, using it
+	return e.pushDataToExporters(ctx, td, e.traceExporters[value])
+}
+
+func (e *processorImp) GetCapabilities() component.ProcessorCapabilities {
+	return component.ProcessorCapabilities{MutatesConsumedData: true}
```

Comment:
Does this processor mutable data anywhere? I can't see it in the code.

Reply:
You are right, it does *not*.